### PR TITLE
Upgrade to Node 16

### LINKED ISSUE #1924: Upgrade to Node 16
**What needs to be done? Why does it need to be done?**
A clear and concise description of what task needs to be accomplished and why.

**Additional context**
Add any other context about the task here.

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

Upgrades the app to Node 16. This depends on the [web-libraries package upgrade](https://github.com/Recidiviz/web-libraries/pull/107) being published.

To support the upgrade, there were a few things that needed to be changed:
- `CoreConstants.scss` needed to change to `CoreConstants.module.scss` so that the built in module support in Node 16 could continue to allow the sharing of variables between css and js.
- All Cucumber tests are changed to be async, because [@wdio/sync is no longer supported in Node 16](https://github.com/webdriverio/webdriverio/issues/7704)
- All react-scripts tests must be run with `--resetMocks=false`  because with [react-scripts v4+, `restMocks` is true by default](https://github.com/facebook/create-react-app/issues/10571).
- The firestore tests are now run with `--env=node` instead of the now default jsdom
- Removed `yarn.lock` from the `.cloudignore` file to ensure that the package is built using yarn during deploy

I have deleted my node modules several times throughout this process. But the last time I installed packages on `main` then ran `yarn install` on this branch I did have to delete my node modules. Let me know if you are able to do this upgrade without deleting.

Both the BE and FE have been deployed to staging from this branch.

QA:
- [x] Login/Logout
- [x] Lantern tenants
- [x] Core tenant
- [x] Pathways
- [x] Workflows
- [x] Mobile
- [x] IE11
- [x] Deploy BE
- [x] Deploy FE

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #1924

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [x] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by macfarlandian** [comment]
I also had to delete my `node_modules` FWIW ... though I think that is to be expected on a major version bump anyway so I would not sweat it

**Review by macfarlandian — APPROVED** [review]
🙌🏻  looks great to me and seems to be running fine both locally and in staging.

In case it's of interest to anyone, to get the tests working in my VS Code integration I also had to change my `nvm default` alias to `16.x` ... the test scripts worked fine as-is once I did that (except for `test-firestore` but that didn't work before either)

**Comment by jovergaag** [comment]
> except for `test-firestore` but that didn't work before either

I've never been able to get `test-firestore` to work in VS Code either.

File: src/core/bars/PerMonthBarChart.js [hunk 1]
```diff
@@ -51,7 +51,10 @@ export const prepareDataGroupedByMonth = (metricType, bars) => (data) => {
   const { year, month } = data;
 
   const monthCounts = reduce(
-    (acc, { key }) => ({ ...acc, [key]: Number(data[key]) }),
+    (acc, { key }) => {
+      const { key: d } = data;
```
Comment (Catacola) [inline src/core/bars/PerMonthBarChart.js:55]: I believe the way this is currently written, `d` will be same same on every loop. If I'm not mistaken, this is equivalent to writing `const d = data.key`. Before, it was accessing `data[key]`. What issue is being addressed here?

File: src/core/bars/PerOfficerBarChart.js [hunk 2]
```diff
@@ -70,7 +70,10 @@ const prepareDataGroupedByOffice = (bars, metricType) => (data) => {
   const officerId = numberFromOfficerId(data.officer_external_id);
 
   const countsByType = reduce(
-    (counts, { key }) => ({ ...counts, [key]: toInteger(data[key]) }),
+    (counts, { key }) => {
+      const { key: d } = data;
+      return { ...counts, [key]: toInteger(d) };
+    },
```
Comment (Catacola) [inline src/core/bars/PerOfficerBarChart.js:76]: I have the same comment here.

**Review by Catacola — COMMENTED** [review]
For a 3.5k-line and 91-file diff, that could have been much worse to review.

I have one minor comment around a test change, but otherwise this looks great to me. Thanks for taking on a big change like this.

File: src/core/bars/PerMonthBarChart.js (hunk 1 above)
Comment (jovergaag) [reply src/core/bars/PerMonthBarChart.js:55]: The new version of ESLint was complaining about it. This was the only way to make lint happy. Same with the next comment.

File: src/cucumber/pages/lanternPage.js [hunk 3]
```diff
@@ -6,60 +6,72 @@ class LanternPage extends Page {
     super.open(`${browser.config.baseUrl}/community/revocations`);
   }
 
-  get lanternLayout() {
-    return $(".LanternLayout");
+  async lanternLayout() {
+    const el = await $(".LanternLayout");
+    return el;
```
[the remark this answers is not in the document]
Comment (jovergaag) [reply src/cucumber/pages/lanternPage.js]: I agree with you though that this isn't as clean as it should be. I went back and cleaned it up.

File: src/core/bars/PerMonthBarChart.js (hunk 1 above)
Comment (Catacola) [reply src/core/bars/PerMonthBarChart.js:55]: If that's the case, then I think the second line needs to be the following to maintain what the code was originally doing
```
      const { [key]: d } = data;
```
or 
```
      const d = data[key];
```

Comment (jovergaag) [reply src/core/bars/PerMonthBarChart.js:55]: Oh you are totally right - good catch. Those charts (ND Core dashboard) were indeed broken, but are fixed now. Thank you!

Comment (jovergaag) [reply src/core/bars/PerMonthBarChart.js]: @Catacola Ready for another look.

**Pull request merged**